src/app/checklist/app-updates/app-updates.service.ts

```diff
@@ -0,0 +1,57 @@
+import { ApplicationRef, Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material';
+import { SwUpdate } from '@angular/service-worker';
+import { fromEvent, merge, Observable, of } from 'rxjs';
+import { first, mapTo, switchMap, tap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AppUpdatesService {
+  constructor(public snackbar: MatSnackBar, private serviceWorkerUpdates: SwUpdate, private appRef: ApplicationRef) {}
+
+  checkAppStability(): Observable<Boolean> {
+    const appIsStable = this.appRef.isStable.pipe(first(isStable => isStable === true));
+    return appIsStable;
+  }
+
+  checkUpdatesAuto() {
+    this.serviceWorkerUpdates.activated.subscribe(() => {
+      this.snackbar.open('App updated', 'OK', { duration: 2000 });
+    });
+
+    this.serviceWorkerUpdates.available
+      .pipe(
+        switchMap(() => {
+          const updateSnackbar = this.snackbar.open('Update available!', 'Update Now');
+          return updateSnackbar.onAction().pipe(tap(() => updateSnackbar.dismiss()));
+        })
+      )
+      .subscribe(() => window.location.reload());
+  }
+
+  checkUpdatesManually() {
+    const updateCheckSnackbar = this.snackbar.open(`Checking for updates...`);
+
+    this.serviceWorkerUpdates
```

Comment:
With the chance of nit-picking. I would prefer to wrap this inside an Observable `from(this.serviceWorkerUpdates`. Just keeping a single async paradigm is what I prefer.

Reply:
@d3lm @KwintenP Ok, I just made the commit, what do you think, everything alright?